src/pybind/mgr/localpool/module.py

```diff
@@ -56,6 +73,106 @@ def notify(self, notify_type, notify_id):
         if notify_type == 'osd_map':
             self.handle_osd_map()
 
+    def get_highest_rule_id(self):
+        crush = self.get('osd_map_crush')
+        crush_rules = crush['rules']
+        id = 1
+        for rule in crush_rules:
+            if rule['rule_id'] > id:
+                id = rule['rule_id']
+
+        return id
+    def create_crush_affined_rule(self, name, region, pd, sd, td):
+        rule_template = Template(""" rule ${name} {
+    id ${rule_id}
+    type replicated
+    min_size 1
+    max_size 10
+    step take ${region}${primary_domain}
+    step chooseleaf firstn 1 type host
+    step emit
+    step take ${region}${secondary_domain}
+    step chooseleaf firstn 1 type host
+    step emit
+    step take ${region}${tertiary_domain}
+    step chooseleaf firstn 1 type host
+    step emit
+}""")
+        
+        tempfd, crushmap_filename = mkstemp(dir='/tmp')
+
+        editable_crushmap_filename = crushmap_filename+".editable"
```

Comment:
lots of style nits...
- space before and after operators like + or = (for assignment, but not for named args)
- space after : in a dict
- space after commas in lists
- space after comment # character